stress/shell/src/main/java/alluxio/stress/cli/MultiOperationStressMasterBench.java

```diff
@@ -0,0 +1,502 @@
+                      op));
+            }
+          }
+        }
+        if (!prepareFs.isDirectory(basePath)) {
+          throw new IllegalStateException(String
+              .format("base path (%s) must be a directory for operation (%s)", basePath,
+                  op));
+        }
+      }
+    }
+
+    if (mParameters.mClientType == FileSystemClientType.ALLUXIO_HDFS) {
+      throw new RuntimeException(
+          "ALLUXIO HDFS Compatible API is not support for multi operations stress master bench");
+    }
+
+    // set hdfs conf for all test clients
+    LOG.info("Using ALLUXIO Native API to perform the test.");
+    InstancedConfiguration alluxioProperties = alluxio.conf.Configuration.copyGlobal();
+    alluxioProperties.set(
+        PropertyKey.USER_FILE_WRITE_TYPE_DEFAULT, mParameters.mWriteType, Source.RUNTIME);
+    mCachedNativeFs = new alluxio.client.file.FileSystem[mParameters.mClients];
+    for (int i = 0; i < mCachedNativeFs.length; i++) {
+      mCachedNativeFs[i] = alluxio.client.file.FileSystem.Factory
+          .create(alluxioProperties);
+    }
+
+    for (int i = 0; i < mParameters.mOperations.length; ++i) {
+      if (mParameters.mOperations[i] == Operation.CRURD
+          || mParameters.mOperations[i] == Operation.CREATE_DELETE_FILE) {
+        AlluxioURI uri = new AlluxioURI(
+            PathUtils.concatPath(mParameters.mBasePaths[i], mFilesDir, mBaseParameters.mId));
+        if (mCachedNativeFs[0].exists(uri)) {
+          mCachedNativeFs[0].delete(uri, DeletePOptions.newBuilder().setRecursive(true).build());
+        }
+        mCachedNativeFs[0].createDirectory(uri, CreateDirectoryPOptions.getDefaultInstance());
+      }
+    }
+
+    // caonima
+  }
+
+  @Override
+  @SuppressFBWarnings("BC_UNCONFIRMED_CAST")
+  protected BenchThread getBenchThread(BenchContext context, int index) {
+    if (mParameters.mClientType == FileSystemClientType.ALLUXIO_NATIVE) {
+      return new AlluxioNativeBenchThread(context,
+          mCachedNativeFs[index % mCachedNativeFs.length], index);
+    }
+    throw new RuntimeException(
+        "ALLUXIO HDFS and POSIX API is not support for multi operations stress master bench");
+  }
+
+  private RateLimiter createUnlimitedRateLimiter() {
+    return RateLimiter.create(Double.MAX_VALUE);
+  }
+
+  @Override
+  @SuppressFBWarnings("BC_UNCONFIRMED_CAST")
+  protected StressMasterBenchBase<
+      MultiOperationMasterBenchTaskResult, MultiOperationMasterBenchParameters>.BenchContext
+      getContext() throws IOException, AlluxioException {
+    final BenchContext benchContext;
+    if (mParameters.mTargetThroughputs == null) {
+      RateLimiter[] rateLimiters = new RateLimiter[mParameters.mOperations.length];
+      for (int i = 0; i < mParameters.mTargetThroughputs.length; i++) {
+        rateLimiters[i] = createUnlimitedRateLimiter();
+      }
+      Arrays.fill(rateLimiters, createUnlimitedRateLimiter());
```

Comment:
remove this line otherwise the elements are overwritten.

Reply:
done